RentACarWebAPI/Services/ClientService.cs

```diff
@@ -13,31 +14,31 @@ public ClientService(IClientRepository clientRepository) : base(clientRepository
             _clientRepository = clientRepository;
         }
 
-        public override Client Create(Client newClient)
+        public override async Task<Client> CreateAsync(Client newClient)
         {
-            var clientExists = _clientRepository.DniExists(newClient);
+            var clientExists = await _clientRepository.DniExistsAsync(newClient);
             if (clientExists) return null;
 
-            var clientCreated = base.Create(newClient);
+            var clientCreated = await base.CreateAsync(newClient);
 
             return clientCreated;
         }
 
-        public override Client Update(Client newClient, int id)
+        public override async Task<Client> UpdateAsync(Client newClient, int id)
         {
-            var existingClient = _clientRepository.Get(id);
+            var existingClient = await _clientRepository.GetAsync(id);
```

Comment:
Can i just check if client exists? Or there is the possibility that the client Id exists and returns null anyway? (i think not, but im not sure)

Reply:
if you want you can only check if the ID exists, it depends on the context (whether you need the entire client object or not)